src/CameraEnvironmental.h

```diff
@@ -3,5 +3,46 @@
 #include "ICamera.h"
 #include "ray.h"
 
-// --- IMPLENET class CCameraEnvironmental ---
+#define _USE_MATH_DEFINES
+#include <math.h>
+// --- IMPLEMENT class CCameraEnvironmental ---
 // --- PUT YOUR CODE HERE ---
+
+class CCameraEnvironmental : public ICamera
+{
+public:
+	CCameraEnvironmental(Size resolution, const Vec3f& pos, float angle)
+		: ICamera(resolution)
+		, m_pos(pos)
+	{
+	}
+	virtual ~CCameraEnvironmental(void) = default;
+
+	virtual void InitRay(Ray& ray, int x, int y) override
+	{
+		// --- PUT YOUR CODE HERE ---
+		float dx = x;
+		float dy = y;
+
+		auto height = getResolution().height;
+		auto width = getResolution().width;
+
+		double theta = 2 * M_PI * dx / height;
+		double phi = acos(1 - 2 * dy / width);
```

Comment:
Here would be enough to have phi changing from 0 to Pi. arccosine is not needed.